pkgs/development/python-modules/psycopg2/default.nix

```diff
@@ -65,7 +65,7 @@ buildPythonPackage rec {
     "psycopg2"
   ];
 
-  disallowedReferences = lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [ buildPackages.postgresql ];
+  disallowedReferences = lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [ buildPackages.libpq ];
```

Comment:
> This means checking for `libpq` references here is useless.

Yeah that is unfortunate. I guess you could have an alias pointing to the actual postgres package like `libpq.postgres` -> `postgres_latest`, but not sure how I feel about that tbh. It's easy to miss this subtle issue if you're not aware of how `libpq` works.

> But since `libpq` always depends on the latest postgresql package it's also useless to check for `postgresql` instead - because we're actually using `postgresql_16`. And we sure don't want to pin a specific version here, but have libpq be on latest all the time.

Maybe instead of symlinking `lib` and `dev` to `libpq` we could copy them and then patch all references like `postgresql.lib` -> `$out` and `postgresql.dev` -> `$dev`. But that's even more hacky and I'm not sure if it's worth the extra complexity. Just putting it out there.

> I think this is actually worse than having a `libpq` package, which only pretends to be lib-only, but is in fact a full `postgrest_latest` package. The use-case should still be clear from the `libpq` name alone.

It's only a full postgres package at build time. If you only reference a binary cached `libpq`, it won't actually download any postgresql server binaries.

Reply:
Alternatively just `libpq = postgresql_latest.override { systemdSupport = false;  # ... and other unnecessary stuff for libpq }` and remove/rearrange the outputs a bit.

I'd like `libpq` to just cross-compile and afaik stuff like systemd doesn't like to be cross-compiled. I'll test this as soon as I bootstrap the cross compiler.

Reply:
For example something like this:
```nix
libpq = (postgresql_latest.override { systemdSupport = false; }).overrideAttrs (old: {
  meta = old.meta // { outputsToInstall = [ "lib" ]; };
});
```

This way the default output only includes `lib`:
```
nix build -f . pkgsCross.aarch64-multiplatform.libpq -L --print-out-paths
/nix/store/gj14qvx3194csxg3bcdc4bmsdrl6kqzn-postgresql-aarch64-unknown-linux-gnu-16.2-lib
```

Unless it's included as an build input, in which case `dev` is still available and included.